include/pybind11/pybind11.h

```diff
@@ -1169,90 +1169,107 @@ class class_ : public detail::generic_type {
-                def("__or__", [](const Type &value, const Type &value2) { return (Scalar) value | (Scalar) value2; });
-                def("__xor__", [](const Type &value, const Type &value2) { return (Scalar) value ^ (Scalar) value2; });
+                cls
+                .def("__lt__", [](const Type &value, Scalar value2) { return (Scalar) value < value2; })
+                .def("__gt__", [](const Type &value, Scalar value2) { return (Scalar) value > value2; })
+                .def("__le__", [](const Type &value, Scalar value2) { return (Scalar) value <= value2; })
+                .def("__ge__", [](const Type &value, Scalar value2) { return (Scalar) value >= value2; })
+                .def("__invert__", [](const Type &value) { return ~((Scalar) value); })
+                .def("__and__", [](const Type &value, Scalar value2) { return (Scalar) value & value2; })
+                .def("__or__", [](const Type &value, Scalar value2) { return (Scalar) value | value2; })
+                .def("__xor__", [](const Type &value, Scalar value2) { return (Scalar) value ^ value2; })
+                .def("__rand__", [](const Type &value, Scalar value2) { return (Scalar) value & value2; })
+                .def("__ror__", [](const Type &value, Scalar value2) { return (Scalar) value | value2; })
+                .def("__rxor__", [](const Type &value, Scalar value2) { return (Scalar) value ^ value2; })
+                .def("__and__", [](const Type &value, const Type &value2) { return (Scalar) value & (Scalar) value2; })
+                .def("__or__", [](const Type &value, const Type &value2) { return (Scalar) value | (Scalar) value2; })
+                .def("__xor__", [](const Type &value, const Type &value2) { return (Scalar) value ^ (Scalar) value2; });
             }
         }
-        def("__hash__", [](const Type &value) { return (Scalar) value; });
+        cls
+        .def("__hash__", [](const Type &value) { return (Scalar) value; })
         // Pickling and unpickling -- needed for use with the 'multiprocessing' module
-        def("__getstate__", [](const Type &value) { return pybind11::make_tuple((Scalar) value); });
-        def("__setstate__", [](Type &p, tuple t) { new (&p) Type((Type) t[0].cast<Scalar>()); });
+        .def("__getstate__", [](const Type &value) { return pybind11::make_tuple((Scalar) value); })
+        .def("__setstate__", [](Type &p, tuple t) { new (&p) Type((Type) t[0].cast<Scalar>()); });
     }
 
     /// Export enumeration entries into the parent scope
-    enum_& export_values() {
+    enum_ export_values() && {
         for (const auto &kv : m_entries)
             m_parent.attr(kv.first) = kv.second;
-        return *this;
+        return std::move(*this);
     }
 
     /// Add an enumeration entry
-    enum_& value(char const* name, Type value) {
+    enum_ value(char const* name, Type value) && {
```

Comment:
Well, if you return a reference, you won't be able to call `.into_class()` which needs an rvalue (which is an important point, in order to avoid all kinds of edge cases). Taking an rvalue `*this` here, disable copy ctor and returning by value basically helps ensure we never copy it (accidentally or not) and always move. The usual chaining-style enum definitions still all work as expected.